cli_tools/common/image/importer/inflater.go

```diff
@@ -47,88 +48,37 @@ type shadowTestFields struct {
 func newInflater(request ImageImportRequest, computeClient daisyCompute.Client, storageClient domain.StorageClientInterface,
 	inspector imagefile.Inspector, logger logging.Logger) (Inflater, error) {
 
-	di, err := newDaisyInflater(request, inspector, logger)
-	if err != nil {
-		return nil, err
-	}
-
 	if isImage(request.Source) {
-		return di, nil
-	}
-
-	if isShadowTestFormat(request) {
-		return &shadowTestInflaterFacade{
-			mainInflater:   di,
-			shadowInflater: createAPIInflater(request, computeClient, storageClient, logger, true),
-			logger:         logger,
-		}, nil
+		return newDaisyInflater(request, imagefile.Metadata{}, logger)
 	}
 
-	return &inflaterFacade{
-		apiInflater:   createAPIInflater(request, computeClient, storageClient, logger, false),
-		daisyInflater: di,
-		logger:        logger,
-	}, nil
-}
-
-func isShadowTestFormat(request ImageImportRequest) bool {
-	// TODO: process VHD/VPC differently
-	return true
-}
-
-// inflaterFacade implements an inflater using other concrete implementations.
-type inflaterFacade struct {
-	apiInflater   Inflater
-	daisyInflater Inflater
-	logger        logging.Logger
-}
+	deadline, cancelFunc := context.WithDeadline(context.Background(), time.Now().Add(inspectionTimeout))
+	defer cancelFunc()
+	metadata, err := inspector.Inspect(deadline, request.Source.Path())
 
-func (facade *inflaterFacade) Inflate() (persistentDisk, shadowTestFields, error) {
-	var pd persistentDisk
-	var tf shadowTestFields
-	var err error
-
-	//  Try with API inflater.
-	pd, tf, err = facade.apiInflater.Inflate()
-	if err == nil {
-		facade.logger.Metric(&pb.OutputInfo{
-			InflationType:   "api_success",
-			InflationTimeMs: []int64{tf.inflationTime.Milliseconds()},
-		})
-		return pd, tf, err
+	if err != nil {
+		logger.Debug(fmt.Sprintf("Error while inspecting disk file. "+
+			"The daisy inflation worker may need to resize the target PD. Error=%s", err))
 	}
 
-	if !isCausedByUnsupportedFormat(err) {
-		facade.logger.Metric(&pb.OutputInfo{
-			InflationType:   "api_failed",
-			InflationTimeMs: []int64{tf.inflationTime.Milliseconds()},
-		})
-		return pd, tf, err
+	di, err := newDaisyInflater(request, metadata, logger)
+	if err != nil {
+		return nil, err
 	}
 
-	// Retry with daisy inflater.
-	pd, tf, err = facade.daisyInflater.Inflate()
-	if err == nil {
-		facade.logger.Metric(&pb.OutputInfo{
-			InflationType:   "qemu_success",
-			InflationTimeMs: []int64{tf.inflationTime.Milliseconds()},
-		})
-		return pd, tf, err
+	if !request.FormatsWithParallelInflation[metadata.FileFormat] {
+		return di, nil
 	}
 
-	facade.logger.Metric(&pb.OutputInfo{
-		InflationType:   "qemu_failed",
-		InflationTimeMs: []int64{tf.inflationTime.Milliseconds()},
-	})
-	return pd, tf, err
-}
-
-func (facade *inflaterFacade) Cancel(reason string) bool {
-	// No need to cancel apiInflater.
-	return facade.daisyInflater.Cancel(reason)
+	return &shadowTestInflaterFacade{
```

Comment:
I think we should still have the gate to determine whether to return &inflaterFacade to use PD API.

Reply:
+1

We should have a property to easily switch between daisy and API inflaters. This will make our lives easier when we do need to switch and there will be less risk of errors.

Reply:
Switching to a quick implementation that disables the PD API